File: board-room/src/main/java/net/stzups/board/data/objects/canvas/object/objects/Shape.java
package net.stzups.board.data.objects.canvas.object.objects;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.canvas.object.CanvasObject;

public class Shape extends CanvasObject {
    private int width;
    private int height;

    public Shape(ByteBuf byteBuf) {
        super(byteBuf);
        width = byteBuf.readUnsignedShort();
        height = byteBuf.readUnsignedShort();
    }

    public void serialize(ByteBuf byteBuf) {
        byteBuf.writeShort((short) width);
        byteBuf.writeShort((short) height);
    }
}

File: frontend/html/scripts/canvas/canvasObjects/Shape.js
import CanvasObject from "../CanvasObject.js";
import {CanvasObjectType} from "../CanvasObjectType.js";

export default class Shape extends CanvasObject {
    constructor(reader) {
        super(CanvasObjectType.SHAPE, reader);
        this.length = reader.readUint16();
        this.width = reader.readUint16();
    }

    serialize(writer) {
        super.serialize(writer);
        writer.writeInt16(this.length);
        writer.writeInt16(this.width);
    }
}